### PR TITLE
Temp c UI

### PR DESCRIPTION
**Pull request by ShreenidhiHD**

### CODE REVIEW

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Common/ValidationType.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    public enum ValidationType
+    {
+        MinPasswordLength =1,
+        Email=2,
+        AdharLength=3,
+        MobileNumberLength=1003,
+        Year=1004,
+        Name=1005,
+        IsRequired=1006
+    }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Common/ValidationType.cs:16]: validation type enum value should be in squence of string constant value not like 1, 2, 3, 1003 etc.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/Common/ICommonService.cs [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-        SampleModel Get(int id);
-        RequestResult<int> Add(SampleModel sample);
-        RequestResult<int> Update(int id, SampleModel sample);
+        RequestResult<bool> Add(RecordModel record);
+        RequestResult<bool> Update(RecordModel record);
         bool Delete(int id);
-        RequestResult<int> AddCollection(List<SampleModel> sample);
+        RecordModel GetUiPageMetadata(int uiPageId);
+        RecordsModel GetRecords();
+        RecordModel GetRecordById(int recordId);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/Common/ICommonService.cs:18]: proper comments of interface methods

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/RecordModel.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    public class RecordModel : Entity
+    {
+        #region Public Properties
+        public int UiPageId { get; set; }
+
+        public List<UiPageMetadataModel> Fields { get; set; }
+
+        public List<UiPageDataModel> FieldValues { get; set; }
+        #endregion
+    }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/RecordModel.cs:18]: put proper comments.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/RecordsModel.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace TestingAndCalibrationLabs.Business.Core.Model
+{
+    public class RecordsModel : Entity
+    {
+        public int UiPageId { get; set; }
+        public List<UiPageMetadataModel> Fields { get; set; }
+        public Dictionary<int, List<Core.Model.UiPageDataModel>> FieldValues { get; set; }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/RecordsModel.cs:11]: put proper comments

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Sample/SampleModel.cs [hunk 5]
```diff
@@ -6,8 +6,8 @@
     public class SampleModel : Entity
     {
         #region Public Properties
-        public string Name { get; set; }
-        public string Description { get; set; }
+        public string UserName { get; set; }
+        public string Password { get; set; }
         #endregion
     }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Sample/SampleModel.cs:12]: why this change is required

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -3,13 +3,18 @@
 
 <head>
     <meta charset="utf-8" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>TESTING AND CALIBRATION LABS</title>
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Shared/_Layout.cshtml:7]: what is this meta change and it is not even closed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Index.cshtml [hunk 7]
```diff
@@ -1,52 +1,46 @@
...
         }
-        @using (Html.BeginForm("Index", "Home", new { pageIndex = @ViewBag.PreviousPage }, FormMethod.Post))
+     @*   @using (Html.BeginForm("Index", "Home", new { pageIndex = @ViewBag.PreviousPage }, FormMethod.Post))
         {
             <button class="btn-primary">Previous</button>
         }
         @using (Html.BeginForm("Index", "Home", new { pageIndex = @ViewBag.nextPage }, FormMethod.Post))
         {
             <button class="btn-primary">Next</button>
-        }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Index.cshtml:50]: why this section is still there? i think we can make this view as common instead of sample.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Edit.cshtml [hunk 8]
```diff
@@ -1,36 +1,122 @@
...
 <div>
     <a asp-action="Index">Back to List</a>
-</div>
\ No newline at end of file
+</div>
+ <div>
+        <div class="d-flex justify-content-center fixed-bottom">
+            <div class="alert alert-success" role="alert" id="snackbarsuccess">
+               
+             <div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 MuiAlert-root MuiAlert-filledSuccess MuiAlert-filled css-1lgz3mj" role="alert"><div class="MuiAlert-icon css-1l54tgj"><svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1cw4hi4" focusable="false" aria-hidden="true" viewBox="0 0 24 24" data-testid="SuccessOutlinedIcon"><path d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"></path></svg></div><div class="MuiAlert-message css-1xsto0d">This is a success message!</div></div>
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Edit.cshtml:46]: why snackbar is hard coded.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Details.cshtml [hunk 9]
```diff
@@ -30,3 +30,4 @@
     @Html.ActionLink("Edit", "Edit", new {  id = Model.Id }) |
     <a asp-action="Index">Back to List</a>
 </div>
+*@
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Details.cshtml:33]: is this change even needed anymore.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs [hunk 10]
```diff
@@ -62,7 +76,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             {
                 endpoints.MapControllerRoute(
                     name: "default",
-                    pattern: "{controller=Home}/{action=login}/{id?}");
+                    pattern: "{controller=Sample}/{action=Index}/{id?}");
             });
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:80]: plz don't check-in the default route.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs [hunk 11]
```diff
@@ -28,12 +33,21 @@ public void ConfigureServices(IServiceCollection services)
...
             
             //Services
+            services.AddScoped<ICommonService, SampleService>();
             services.AddScoped<ISurveyService, SurveyService>();
             services.AddScoped<IEmailService, EmailService>();
             services.AddScoped<ISampleService, SampleService>();
             
             //Repository
             services.AddScoped<IConnectionFactory, ConnectionFactory>();
+            //services.AddScoped<IGenericRepository<SampleModel>, GenericRepository<SampleModel>>();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:43]: commented code?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Shared/_Layout.cshtml (hunk 6 above)
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Shared/_Layout.cshtml:7]: That is the right way of written meta there is no tag close for meta . Reference Official bootstrap website https://getbootstrap.com/docs/5.0/getting-started/introduction/ 


Example from BS5 website.

`<!doctype html>
<html lang="en">
  <head>
    
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">

    <title>Hello, world!</title>
  </head>
  <body>
    <h1>Hello, world!</h1>

    

    
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>

    
    
  </body>
</html>`

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Index.cshtml (hunk 7 above)
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Index.cshtml:50]: Sample is good I think View can not make common right?

Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Index.cshtml:50]: Section removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Edit.cshtml (hunk 8 above)
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Sample/Edit.cshtml:46]: i didn't get

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs (hunk 10 above)
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:80]: okay

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs (hunk 11 above)
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:43]: Removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Shared/_Layout.cshtml [hunk 12]
```diff
@@ -77,7 +84,7 @@
     <div class="">
         <main role="main" class="">
             @RenderBody()
-        </main>
+       <!-- </main> -->
     </div>
```
[the remark this answers is not in the document]
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Views/Shared/_Layout.cshtml:88]: updated but some of these changes are came while removing merge conflicts

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Sample/SampleModel.cs (hunk 5 above)
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Sample/SampleModel.cs:12]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Common/ValidationType.cs (hunk 1 above)
Comment (ShreenidhiHD) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Common/ValidationType.cs:16]: I will update it in script then  and here too
